pkg/kubernetes/cluster.go

```diff
@@ -0,0 +1,223 @@
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	serverConfigFile = "server.yaml"
+	agentConfigFile  = "agent.yaml"
+
+	tokenKey        = "token"
+	cniKey          = "cni"
+	cniDefaultValue = image.CNITypeCilium
+	serverKey       = "server"
+	tlsSANKey       = "tls-san"
+)
+
+type Cluster struct {
+	// Initialiser is the hostname of the initialiser node.
+	// Defaults to the first configured server if not explicitly selected.
+	Initialiser string
```

Comment:
Nit: In the definition, `Initialiser` is a boolean, but here it's the hostname. There shouldn't be any technical confusion since they are different types, though it might be a bit more clear to new devs if this was `InitialiserHostname` (esp since it matches the pattern of `InitialiserConfig` below).